src/libraries/ArrayUtil.sol

```diff
@@ -479,34 +470,37 @@ library ArrayUtil {
     @dev this function modifies memory x IN PLACE. Fills x with value v
     */
     function fill(int256[] memory x, int256 v) internal pure {
-        for (uint256 i = 0; i < x.length; i++) {
+        for (uint256 i = 0; i < x.length; ) {
             x[i] = v;
+
+            unchecked {
+                ++i;
+            }
         }
     }
 
     /*
     @dev modifies memory a IN PLACE. Populates a starting at index z with values from b.
     */
-    function populate(
-        uint256[] memory a,
-        uint256[] memory b,
-        uint256 z
-    ) internal pure {
-        for (uint256 i = 0; i < a.length; i++) {
+    function populate(uint256[] memory a, uint256[] memory b, uint256 z) internal pure {
+        for (uint256 i = 0; i < a.length; ) {
             a[z + i] = b[i];
+
+            unchecked {
+                ++i;
+            }
         }
     }
 
     /*
     @dev modifies memory a IN PLACE. Populates a starting at index z with values from b.
     */
-    function populate(
-        int256[] memory a,
-        int256[] memory b,
-        uint256 z
-    ) internal pure {
-        for (uint256 i = 0; i < a.length; i++) {
+    function populate(int256[] memory a, int256[] memory b, uint256 z) internal pure {
+        for (uint256 i = 0; i < a.length; ) {
             a[z + i] = b[i];
+            unchecked {
```

Comment:
nit picky, but can you add a space here. It seems like we are mostly putting a new line before `unchecked` blocks.

Reply:
Looks like there a few places where this comments applies.